gamemodes/terrortown/entities/entities/ttt_logic_role.lua

```diff
@@ -24,35 +26,46 @@ function ENT:KeyValue(key, value)
 			value = _G[value] or value
 		end
 
-		self.Role = tonumber(value)
+		self.checkingRole = tonumber(value)
 
-		if not self.Role then
+		if not self.checkingRole then
 			ErrorNoHalt("ttt_logic_role: bad value for Role key, not a number\n")
 
-			self.Role = ROLE_NONE
+			self.checkingRole = ROLE_NONE
 		end
 	end
 end
 
 ---
--- @param string name
--- @param Entity|Player activator
--- @return[default=true] boolean
--- @realm shared
-function ENT:AcceptInput(name, activator)
+-- Called when another entity fires an event to this entity.
+-- @param string name The name of the input that was triggered
+-- @param Entity|Player activator The initial cause for the input getting triggered (e.g. the player who pushed a button)
+-- @param Entity caller The entity that directly triggered the input (e.g. the button that was pushed)
+-- @param string data The data passed
+-- @return[default=true] boolean Return true if the default action should be supressed
+-- @realm server
+function ENT:AcceptInput(name, activator, caller, data)
 	if name == "TestActivator" then
-		if IsValid(activator) and activator:IsPlayer() then
-			local activator_role = (GetRoundState() == ROUND_PREP) and ROLE_INNOCENT or activator:GetBaseRole()
+		if not IsValid(activator) or not activator:IsPlayer() then return end
+
+		---
+		-- @realm server
+		local role, team = hook.Run("TTT2ModifyLogicRoleCheck", activator, self, activator, caller, data)
+		local activatorRole = roles.GetByIndex(role):GetBaseRole()
```

Comment:
A check / assertion that this role index is actually available would be useful for other developers when debugging

Reply:
Didn't we decide that we want "hard fails"?

Reply:
No rather fail early. I just think an assertion in a place that responds to others developing with such hooks makes more sense

Reply:
Hm, `roles.GetByIndex` never fails because it returns `ROLE_NONE` as a fallback. So I don't think an assert is needed here

Reply:
Shouldn't it be `ROLE_INNOCENT` by default regarding your following comment?

Reply:
I don't get what you mean here, sorry. `roles.GetByIndex` returns `ROLE_NONE`, not `ROLE_INNOCENT`

Reply:
The default TTT and default maps are using `ROLE_INNOCENT` as well. If the reason to keep `TEAM_INNOCENT` is to have backward compatibility, then it should be `ROLE_INNOCENT` here as well

Reply:
So you mean I should change the fallback `ROLE_NONE` here to `ROLE_INNOCENT`?

Reply:
not exactly. The fallback of the function still should be ROLE_NONE as most of TTT role add-ons are done by our community and so are up-to-date. BUT the fallback / default value of the return here should be `ROLE_INNOCENT`, if I did not got things wrong.